lib/src/http.dart

```diff
@@ -311,13 +311,36 @@ extension Throwing on http.BaseResponse {
   }
 }
 
-extension SyncSending on http.Client {
-  /// Sends an HTTP request and synchronously returns the response. The regular
-  /// send method on [http.Client], which returns a [http.StreamedResponse], is
-  /// the only method that accepts a request object. This method can be used
+extension RequestSending on http.Client {
+  /// Sends an HTTP request, validates the response headers, and if validation
+  /// is successful, reads the whole response body and returns it.
+  ///
+  /// The send method on [http.Client], which returns a [http.StreamedResponse],
+  /// is the only method that accepts a request object. This method can be used
   /// when you need to send a request object but want a regular response object.
-  Future<http.Response> sendSync(http.BaseRequest request) async {
+  ///
+  /// If false is passed for [throwIfNotOk], the response will not be validated.
+  /// See [http.BaseResponse.throwIfNotOk] extension for validation details.
+  Future<http.Response> fetch(http.BaseRequest request,
```

Comment:
I don't mind, but I have mixed feelings -- mostly because it's easy to call `send()` by accident instead of using `fetch()`.

or rather it's not very obvious that `fetch` is intended to throw something in certain cases. But maybe, this is something we can refactor later.

Reply:
We could do:
```dart

class PubHttpClient {
  Future<http.Response> fetch(http.BaseRequest request);
  Future<http.StreamingResponse> fetchStream(http.BaseRequest request);

  // For the new cases where you really need it.
  http.Client rawClient();
}
```
But maybe, we do this sort of thing later. Now that we're moving `package:oauth2` into `pub` -- we could consider refactoring all of that, and just have a `fetch` method and `fetchStream` method. But it's a lot more refactoring, let's cleanup later.